halo/attest/keeper/keeper.go

```diff
@@ -906,16 +903,37 @@ func (k *Keeper) deleteBefore(ctx context.Context, height uint64) error {
 			return errors.Wrap(err, "value att")
 		} else if att.GetCreatedHeight() > height {
 			return errors.New("query sanity check [BUG]")
+		} else if att.GetChainId() == k.cChainID && att.GetCreatedHeight() > cHeight {
+			// Consensus chain attestations are deleted much later, since they have possible valset update dependencies.
+			continue
 		}
 
 		// Never delete anything after the last approved attestation offset per chain,
 		// even if it is very old. Otherwise, we could introduce a gap
 		// once we start catching up.
 		// Also, don't delete anything if we don't have an approved attestation yet (leave all pending attestations).
-		if latest, ok, err := getLatest(att.XChainVersion()); err != nil {
+		if latest, ok, err := latestOffset(ctx, att.XChainVersion()); err != nil {
 			return err
 		} else if !ok || att.GetBlockOffset() >= latest {
-			continue
+			continue // Skip deleting finalized attestation.
+		}
+
+		var fuzzyDepsFound bool
+		for _, fuzzyVer := range fuzzyDependents(att.XChainVersion()) {
+			// For finalized attestations (with fuzzy dependents), only delete them AFTER all fuzzy attestations have been deleted.
+			// This avoids deleting finalized overrides for current or future fuzzy attestations.
+			earliestFuzzy, ok, err := earliestOffset(ctx, fuzzyVer)
+			if err != nil {
+				return err
+			} else if !ok && att.GetBlockOffset() <= earliestFuzzy {
```

Comment:
I think the logic here is incorrect. I think you want:
- `||` instead of `&&`
- `att.GetBlockOffset() >= earliestFuzzy`

Reply:
inverse yes